apps/taskman/models.py

```diff
@@ -58,12 +75,22 @@ class Task(NullStrFieldsMixin):
     )
     flaw = models.OneToOneField(Flaw, on_delete=models.CASCADE, related_name="task")
 
+    @property
+    def workflow_state(self):
+        for key, value in WORKFLOW_TO_JIRA_STATUS.items():
+
+            if value == (self.status, self.resolution):
+                return key
+        raise KeyError(
+            "Current combination of Jira status and resolution does not met any workflow state mapped."
+        )
+
     def update_in_memory(self, issue):
         """Deserializes a jira issue into task model"""
         fields = issue["fields"]
         self.jira_key = issue["key"]
         self.team_id = (
-            fields[JiraCustomFields.TEAM] if fields[JiraCustomFields.TEAM] else ""
+            fields[JiraCustomFields.TEAM]["id"] if fields[JiraCustomFields.TEAM] else ""
```

Comment:
This would also be better squashed.

Reply:
changed to the correct commit it was created.